File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if(s.equals(t)){
            return true;
        }
        int index = 0;
        for(int i = 0;i<t.length();i++){
            if(index>=s.length()){
                return true;
            }
            if(t.charAt(i)==s.charAt(index)){
                index++;
            }
            if(index>=s.length()){
                return true;
            }
        }
        return false;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int pointer = 0;
        for(int i=0;i<t.size();i++){
            if(pointer==s.size()){
                return true;
            }
            if(t.at(i)==s.at(pointer)){
                pointer++;
            }
        }
        return pointer == s.size();
    }
};